src/component/Timer/index.jsx

```diff
@@ -22,11 +31,43 @@ const Timer = () => {
 
     const reset = () => {
       timeOverSoundAudio.play();
+      const showNotification = () => {
+        let text = null;
+        const img = 'https://source.unsplash.com/1600x900/?nature,water;';
+        if (counter.current === 6) {
+          text = `Great job! Take a long break. You have ${PERIODS.longBrk.secs} minutes.`;
+          (() => new Notification('Pomodoro', {
+            body: text,
+            icon: img
+          }))();
+        } else if (currentPeriod.id === 'short-brk' || currentPeriod.id === 'long-brk') {
+          text = `Time to get back to work! Your next break starts in ${PERIODS.work.secs} minutes.`;
+          (() => new Notification('Pomodoro', {
+            body: text,
+            icon: img
+          }))();
+        } else if (currentPeriod.id === 'work') {
+          text = `Nice work! Take a short break. You have ${PERIODS.shortBrk.secs} minutes.`;
+          (() => new Notification('Pomodoro', {
+            body: text,
+            icon: img
+          }))();
+        }
+      };
+      if (Notification.permission === 'granted') {
+        showNotification();
+      } else if (Notification.permission !== 'denied') {
+        Notification.requestPermission().then((permission) => {
+          if (permission === 'granted') {
+            showNotification();
+          }
+        });
+      }
```

Comment:
I would like you to move all this logic for showing a notification outside of your React component and into the `timer.js` service. That way, you will only have one line of code in here that calls the function defined in `timer.js`. @motaz99 should be able to help you with this since I did this with him before.

Reply:
I don't have any ideas about this, but I will ask motaz and figure it out.